iroha/src/block.rs

```diff
@@ -33,19 +35,7 @@ impl Block {
     }
 
     pub fn hash(&self) -> Hash {
-        use ursa::blake2::{
-            digest::{Input, VariableOutput},
-            VarBlake2b,
-        };
-
-        let bytes: Vec<u8> = self.into();
-        let vec_hash = VarBlake2b::new(32)
-            .expect("Failed to initialize variable size hash")
-            .chain(bytes)
-            .vec_result();
-        let mut hash = [0; 32];
-        hash.copy_from_slice(&vec_hash);
-        hash
+        Crypto::hash(self.into())
```

Comment:
To keep PRs a bit smaller, I think next time you can put crypto refactor into a separate PR from consensus logic.

Reply:
Thanks, will take into account next time.